tsdb/shard.go

```diff
@@ -1739,26 +1747,40 @@ func (fs *MeasurementFieldSet) IsEmpty() bool {
 func (fs *MeasurementFieldSet) Save() (err error) {
```

Comment:
This feels too complicated.

I think it would be better to just have a queue + a goroutine that handles the queue. Each time the goroutine wakes up with something new in the queue, it can:

1) read all the requests to save the MeasurementFieldSet
2) Marshal the latest one
3) If queue has new stuff in it, go to 1
4) Save the MeasurementFieldSet to a file
5) Use the handles from the requests to unblock the goroutines that made the requests

This would also make it easier to set safety valves (e.g. wait only a certain time in the client, then error; every N requests force a save instead of going back to the queue..)